### PR TITLE
CI: Add macOS, Windows x64 and Linux arm64 test support

### PR DESCRIPTION
**Pull request by thrasher-**
# PR Description

Consolidates the existing CI setup to GitHub. Adds macOS, Windows amd64 and Linux arm64 support as we had test issues related to float64 arch precision which PR https://github.com/thrasher-corp/gocryptotrader/pull/1407 corrects. This will now allow us to detect those issues:  https://github.com/thrasher-/gocryptotrader/actions/runs/7190731571/job/19584336028#step:8:4272

NOTE: GitHub has macOS arm64 hosted runners but they cost money: https://github.com/actions/runner-images/issues/8439 so only x64 is supported for now.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run
- [x] CI runners

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions with my changes

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -34,45 +65,53 @@ jobs:
...
-        PSQL_TESTDBNAME: gct_dev_ci
-        PSQL_SSLMODE: disable
+        GOARCH: ${{ matrix.goarch }}
 
     - name: Upload coverage to Codecov
       uses: codecov/codecov-action@v3
 
-  backend-32bit:
-    name: GoCryptoTrader back-end 32-bit with PSQL
+  backend-arm64:
```
Comment (gloriousCode) [inline .github/workflows/tests.yml:112]: For the sake of consistency, 
```suggestion
  back-end (ubuntu-latest, arm64, false, true)
```

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Big improvements. Appveyor was slow + different architecture bugs oversight + This looks like it fixes the `Cannot open: File exists` verbose output for github actions.

Only downside is the extra amount of red X's will appear when I commit code 😭

**Review by gloriousCode — APPROVED** [review]
Changed to approved after my incorrect suggestion

File: .github/workflows/tests.yml (hunk 1 above)
Comment (thrasher-) [reply .github/workflows/tests.yml:112]: You were very close, just had to be set on the name var below so I've pushed a commit to do that 🚀

**Review by shazbert — APPROVED** [review]
LGTM! Those 🦾 errors 😭

**Review by shazbert — APPROVED** [review]
Rebase confirms test fix orderbook. Re-ACK. Nice 🦅

**Review by gloriousCode — APPROVED** [review]
retACK! I ran locally too, the panic in the arm64 github action wasn't reproduced

**Comment by kgantchev** [comment]
Might I make a suggestion? Feel free to use [FlyCI](https://www.flyci.net/)'s M1 and M2 runners. Our runners are on average 2x faster and 2x cheaper than GitHub's **AND** we have a free tier for OSS projects (see below).

## Install Instructrions 

Easily replace your M1 runners: 
```diff
jobs:
 ci:
-    runs-on: macos-latest
+    runs-on: flyci-macos-large-latest-m1
   steps:
   - name: 👀 Checkout repo
     uses: actions/checkout@v4
```

Or try the M2 runners: 
```diff
jobs:
  ci:
-    runs-on: macos-latest
+    runs-on: flyci-macos-large-latest-m2
    steps:
      - name: 👀 Checkout repo
        uses: actions/checkout@v4
 ```

## Pricing  

| Processor | vCPU | RAM (GB) | Storage | Label | Price on FlyCI | Price on GitHub |
|-----------|-------|-----------|----------|-------------------------------|----------------|-----------------|
| M1        | 4    | 7        | 28 GB | flyci-macos-large-latest-m1   | $0.06          | -             |
| M1        | 8    | 14       | 28 GB | flyci-macos-xlarge-latest-m1  | $0.12          | $0.16           |
| M2        | 4    | 7        | 28 GB | flyci-macos-large-latest-m2   | $0.08          | -             |
| M2        | 8    | 14       | 28 GB | flyci-macos-xlarge-latest-m2  | $0.16          | -             |

## 500 mins/month Free for Public Repos

If your repo is public, then FlyCI offers 500 mins/month of free M1 runner usage with the `flyci-macos-large-latest-m1` runner.  

Best Regards,
Kiril Gantchev
CEO and co-founder of [FlyCI](https://flyci.net)

**Pull request merged**